presto-verifier/src/test/java/com/facebook/presto/verifier/checksum/TestChecksumValidator.java

```diff
@@ -387,6 +393,78 @@ public void testRow()
+                ImmutableMap.<String, Object>builder()
+                        .put("map_checksum", new SqlVarbinary(new byte[] {0x1a}))
+                        .put("map_keys_checksum", new SqlVarbinary(new byte[] {0x0b}))
+                        .put("map_values_checksum", new SqlVarbinary(new byte[] {0x0c}))
+                        .put("map_cardinality_sum", 3L)
+                        .build());
+
+        String expectedOutput = "control(checksum: 0a, keys_checksum: 0b, values_checksum: 0c, cardinality_sum: 3) " +
+                "test(checksum: 1a, keys_checksum: 0b, values_checksum: 0c, cardinality_sum: 3)";
+        assertEquals(
+                checksumValidator.getMismatchedColumns(columns, controlChecksum, testChecksum),
+                ImmutableMap.builder()
+                        .put(MAP_COLUMN, new ColumnMatchResult(false, MAP_COLUMN, expectedOutput))
+                        .build());
+
+        // Mismatched map keys & values checksum
+        testChecksum = new ChecksumResult(
+                5,
+                ImmutableMap.<String, Object>builder()
+                        .put("map_checksum", new SqlVarbinary(new byte[] {0x0a}))
+                        .put("map_keys_checksum", new SqlVarbinary(new byte[] {0x1b}))
+                        .put("map_values_checksum", new SqlVarbinary(new byte[] {0x1c}))
+                        .put("map_cardinality_sum", 3L)
+                        .build());
+
+        expectedOutput = "control(checksum: 0a, keys_checksum: 0b, values_checksum: 0c, cardinality_sum: 3) " +
+                "test(checksum: 0a, keys_checksum: 1b, values_checksum: 1c, cardinality_sum: 3)";
+        assertEquals(
+                checksumValidator.getMismatchedColumns(columns, controlChecksum, testChecksum),
+                ImmutableMap.builder()
+                        .put(MAP_COLUMN, new ColumnMatchResult(false, MAP_COLUMN, expectedOutput))
+                        .build());
+
+        // Mismatched map cardinality sum
+        testChecksum = new ChecksumResult(
+                5,
+                ImmutableMap.<String, Object>builder()
+                        .put("map_checksum", new SqlVarbinary(new byte[] {0x0a}))
+                        .put("map_keys_checksum", new SqlVarbinary(new byte[] {0x0b}))
+                        .put("map_values_checksum", new SqlVarbinary(new byte[] {0x0c}))
+                        .put("map_cardinality_sum", 4L)
+                        .build());
+
+        expectedOutput = "control(checksum: 0a, keys_checksum: 0b, values_checksum: 0c, cardinality_sum: 3) " +
+                "test(checksum: 0a, keys_checksum: 0b, values_checksum: 0c, cardinality_sum: 4)";
+        assertEquals(
+                checksumValidator.getMismatchedColumns(columns, controlChecksum, testChecksum),
+                ImmutableMap.builder()
+                        .put(MAP_COLUMN, new ColumnMatchResult(false, MAP_COLUMN, expectedOutput))
+                        .build());
+    }
```

Comment:
You can combine those mismatched cases into one.
Also, add a test case where the key/value are not orderable and hence the `array_sort` function call is not used.